### PR TITLE
Add keyvault deployment, add azure-advanced example

### PR DESCRIPTION
**Pull request by jmspring**

### CODE REVIEW

File: cluster/azure/keyvault/main.tf [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+data "azurerm_client_config" "current" {}
+
+resource "azurerm_key_vault" "keyvault" {
+  name                = "${var.keyvault_name}"
+  location            = "${azurerm_resource_group.keyvault.location}"
+  resource_group_name = "${azurerm_resource_group.keyvault.name}"
+  tenant_id           = "${data.azurerm_client_config.current.tenant_id}"
+
+  sku {
+    name = "standard"
```
Comment (dtzar) [inline cluster/azure/keyvault/main.tf]: Convert to var

File: cluster/azure/keyvault/main.tf [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  }
+
+  network_acls {
+    default_action = "Allow"
+    bypass         = "AzureServices"
+  }
+
+  tags {
+    environment = "Production"
+  }
```
Comment (dtzar) [inline cluster/azure/keyvault/main.tf]: Remove tags or convert to var

File: cluster/environments/azure-advanced/main.tf [hunk 3]
```diff
@@ -0,0 +1,53 @@
+# terraform {
+#    backend "azurerm" {
+#    }
+# }
+
+module "provider" {
```
Comment (dtzar) [inline cluster/environments/azure-advanced/main.tf]: Convert this to the cleaned up code now found in Azure simple in master.

**Comment by dtzar** [comment]
Overall looks good! Look forward to expanding the advanced example.

**Comment by jmspring** [comment]
@dtzar - requested updates made.

**Pull request merged**